src/modules/lists/lists.controller.ts

```diff
@@ -70,4 +71,49 @@ export class ListsController {
     };
   }
 
+  @ApiOperation({ title: 'Updates a given list' })
+  @Put('/:listid')
+  async updateList(@Req() request: Request, @Param('userid') userId: string, @Param('listid') listId: string, @Body() data: UpdateList) {
+    const current: User = await this.usersService.findByAuth0Id(request.user.auth0Id);
+    const user: User = await this.usersService.findById(userId);
+
+    // check if name is provided
+    if (!data.name) {
+      throw new BadRequestException('name not provided');
+    }
+
+    // check if user exists
+    if (!user) {
+      throw new BadRequestException('User not found');
+    }
+
+    // only admins and current user can update lists
+    if (!current._id.equals(user._id) && !current.roles.includes(Role.ADMIN)) {
+      throw new UnauthorizedException('Not authorized to perform this operation');
+    }
+
+    const list: List[] = await this.listsService.findByUserId({userId, listId});
+
+    // check if list exists
+    if (list.length < 1) {
+      throw new BadRequestException('list not found');
+    }
+
+    // check if list is favourite
+    if (list[0].isFavorite) {
```

Comment:
Instead of adding this additional check, I think it will be better to send this as a new fitler to the `findByUserId` method:

```
const list: List[] = await this.listsService.findByUserId({ userId, listId, isFavorite: false });
```

The update `findByUserId` to support this new filter, this way we will only get the none favorite list from the database directly, and we don't have to worry about this additional check

Reply:
cool i will change it